src/response_decoder.ts

```diff
@@ -264,12 +264,12 @@ function decodeMessageHeader(dataResponse: DataReader) {
     throw new Error(`invalid composite header %#02x: ${type}`)
   }
 
-  const nextType = dataResponse.readInt8()
-  decodeFormatCode(dataResponse, nextType)
+  // next, the composite type is encoded as an AMQP uint8
+  dataResponse.readUInt64()
```

Comment:
Are you sure about it?

Reply:
This is also done in the dotnet client 
https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/RabbitMQ.Stream.Client/AMQP/AmqpWireFormattingRead.cs#L389